pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -159,9 +160,11 @@ public HelixServerStarter(String helixClusterName, String zkAddress, Configurati
     LOGGER.info("Using class: {} as the AccessControlFactory", accessControlFactoryClass);
     final AccessControlFactory accessControlFactory;
     try {
-      accessControlFactory = (AccessControlFactory) Class.forName(accessControlFactoryClass).newInstance();
+      accessControlFactory = PluginManager.get().createInstance(accessControlFactoryClass);
     } catch (Exception e) {
-      throw new RuntimeException("Caught exception while creating new AccessControlFactory instance", e);
+      throw new RuntimeException(
+          "Caught exception while creating new AccessControlFactory instance in class " + this.getClass()
```

Comment:
The class name you want to add to the exception is not `this.getClass()`. Instead, it should be the value of `accessControlFactory`. The idea is that the exception message should reflect the remedial action as close as possible (e.g. in this case, it could be a config typo). It may even be useful to put it in quotes, so as to spot extra spaces (e.g. `"... in class ' + accessControlFactoryClass + "'", e)`)

Reply:
Got it. good point.